docker-compose-rule-core/src/test/java/com/palantir/docker/compose/execution/CallbackThenDelegateShutdownStrategyShould.java

```diff
@@ -40,4 +45,29 @@ public void call_callback_then_call_delegate() throws Exception {
         inOrder.verify(delegate).shutdown(dockerCompose, docker);
         inOrder.verifyNoMoreInteractions();
     }
+
+    @Test
+    public void call_delegate_even_when_callback_throws() throws IOException, InterruptedException {
+        ShutdownStrategy delegate = mock(ShutdownStrategy.class);
+        Runnable callback = mock(Runnable.class);
+
+        RuntimeException callbackException = new RuntimeException("exception in callback");
+        doThrow(callbackException).when(callback).run();
+
+        DockerCompose dockerCompose = mock(DockerCompose.class);
+        Docker docker = mock(Docker.class);
+
+        try {
+            ShutdownStrategy.callbackAndThen(callback, delegate).shutdown(dockerCompose, docker);
+            fail("expected exception");
```

Comment:
hm, would have been nicer to use assertj's `assertThatThrownBy`, but didn't want to move all the tests over to using that over hamcrest `assertThat`

Reply:
I think it's acceptable to have both on the classpath and use them in different test classes. No need to write more hamcrest if the we can use the more discoverable/readable assertj.